dynamic-layers/meta-python2/recipes-devtools/clang/libclc_git.bb

```diff
@@ -0,0 +1,51 @@
+DESCRIPTION = "LLVM based OpenCL runtime support library"
+HOMEPAGE = "http://libclc.llvm.org/"
+SECTION = "libs"
+
+require recipes-devtools/clang/clang.inc
+require recipes-devtools/clang/common-source.inc
+
+TOOLCHAIN = "clang"
+
+LIC_FILES_CHKSUM = "file://libclc/LICENSE.TXT;md5=7cc795f6cbb2d801d84336b83c8017db"
+
+inherit cmake pkgconfig pythonnative qemu
+
+DEPENDS_append = " clang-native qemu-native clang"
+
+OECMAKE_SOURCEPATH = "${S}/libclc"
+
+EXTRA_OECMAKE += " \
+				-DCMAKE_CROSSCOMPILING_EMULATOR=${WORKDIR}/qemu-wrapper.sh \
+				-DCMAKE_CXX_FLAGS=-std=c++17 \
+				-DCMAKE_SHARED_LINKER_FLAGS=-std=c++17 \
+				-DCMAKE_EXE_LINKER_FLAGS=-std=c++17 \
+				-Dclc_comp_in:FILEPATH=${OECMAKE_SOURCEPATH}/cmake/CMakeCLCCompiler.cmake.in \
+				-Dll_comp_in:FILEPATH=${OECMAKE_SOURCEPATH}/cmake/CMakeLLAsmCompiler.cmake.in \
+			"
+
+python do_create_qemu_wrapper () {
+    """Creates a small wrapper that invokes QEMU to run some target binaries
+    on the host."""
+    qemu_libdirs = [d.expand('${STAGING_DIR_HOST}${libdir}'),
+                    d.expand('${STAGING_DIR_HOST}${base_libdir}')]
+    qemu_cmd = qemu_wrapper_cmdline(d, d.getVar('STAGING_DIR_HOST', True),
+                                    qemu_libdirs)
+    wrapper_path = d.expand('${WORKDIR}/qemu-wrapper.sh')
+    with open(wrapper_path, 'w') as wrapper_file:
+        wrapper_file.write("""#!/bin/sh
+
+# This file has been generated automatically.
+# It invokes QEMU to run binaries built for the target in the host during the
+# build process.
+
+%s "$@"
+""" % qemu_cmd)
+    os.chmod(wrapper_path, 0o755)
```

Comment:
can we reuse the qemuwrapper from oe-core here ? see `meta/classes/qemu.bbclass` and perhaps see `meta/classes/gtk-doc.bbclass` how it is used

Reply:
I will look into this, too.

Reply:
The current recipe does use `qemu.bbclass`, it's just adds a python task to create the wrapper as done in `meta-browser/recipes-browser/chromium/chromium-gn.inc`
But I have rewritten it in a shell scriptlet much like in `gtk-doc.bbclass`. The difference is that the wrapper script cannot be in `${B}` because `cmake` pretty much cleans out files it doesn't know about, so it needs to be in `${WORKDIR}`.